### PR TITLE
Detective revolver no longer can be turned into a flawless .357 revolver; In this house our gun explodes from excessive pressure

### LINKED ISSUE #82324: Detective revolver can fire 357 with no penalties
## Issue Summary

The detective revolver can bypass the misfire chance and fire 357 rounds with no downsides. Yeehaw

## Round ID:
[226381](https://scrubby.melonmesa.com/round/226381)

## Reproduction:
Unload the revolver fully
Wrench it to 357
load 5 bullets in
Wrench it back (The revolver wont shoot you as one round isn't chambered)
Pull the trigger to cycle the cylinder and load a live round
you now have 5 .357 bullets in your .38 that deal full damage without misfiring.
Become the station's sheriff and obliterate all antagonists (Optional)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/82324

Also cleans up a lot of the code around the issue itself.

## Why It's Good For The Game

You should not be able to modify this revolver while it is loaded whatsoever. I don't know when exactly this broke, but time has clearly not been kind to the feature. 

I think it was actually impossible to reach the self explosion on modifying the gun? The return I added probably prevented it outright. I guess that was broken the entire time. Honestly, I feel like that's going to be a maintainability issue to keep it working, so maybe we just do away with what we never had.

Maybe it was for the very, very unlikely possibility that you could theoretically load in a bullet DURING the do_after? I think I was just sleep deprived. Like right now.

## Changelog
:cl:
fix: Detective revolvers can no longer exist in a quantum state of both being chambered in .38 and .357, without risk of misfire. Detectives, cowboy enthusiasts and reckless federal agents across the Spinward Sector mourn the loss.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -674,46 +674,39 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
...
 
 /obj/item/gun/ballistic/proc/guncleaning(mob/user, obj/item/A)
-	if(misfire_probability == 0)
+	if(misfire_probability == initial(misfire_probability))
 		balloon_alert(user, "it's already clean!")
 		return
 
 	user.changeNext_move(CLICK_CD_MELEE)
-	user.visible_message(span_notice("[user] begins to cleaning [src]."), span_notice("You begin to clean the internals of [src]."))
+	balloon_alert(user, "cleaning.")
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic.dm:682]:
```suggestion
	balloon_alert(user, "cleaning...")
```

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -674,46 +674,39 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
...
 
 	if(do_after(user, 100, target = src))
-		var/original_misfire_value = initial(misfire_probability)
-		if(misfire_probability > original_misfire_value)
-			misfire_probability = original_misfire_value
-			user.visible_message(span_notice("[user] cleans [src] of any fouling."), span_notice("You clean [src], removing any fouling, preventing misfire."))
-			return TRUE
+		misfire_probability = initial(misfire_probability)
+		balloon_alert(user, "fouling cleaned out!")
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic.dm:686]: If this is a success, no exclamation mark. 
```suggestion
		balloon_alert(user, "fouling cleaned out")
```

File: code/modules/projectiles/guns/ballistic.dm [hunk 3]
```diff
@@ -674,46 +674,39 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
...
-			return
-
-		else if(!bolt_locked)
-			to_chat(user, span_notice("You can't get at the internals while the bolt is down!"))
-			return
+	if(!bolt_locked && bolt_type == BOLT_TYPE_LOCKING)
+		balloon_alert(user, "the bolt is in the way!")
+		return
 
 	to_chat(user, span_notice("You begin to tinker with [src]..."))
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic.dm:705]: This one gets balloon'd?

Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic.dm]: sure

**Comment by Acer202** [comment]
"quantum state of both being chambered in .38 and .357"
On this topic... can it be made so .357 chambered guns also accept .38 like in real life? That'd be a fun change.

**Comment by necromanceranne** [comment]
would mean making accepted calibers a list or something, which is doable just kind of fiddly. And, then, you'd have to have stuff like misfires check if the firing caliber is our 'explode the gun' caliber or something. Benefit; we can kill off some procs like ammo swapping entirely. Con; it might be a lot of work for literally just like three guns at most.

**Pull request merged**